src/include/distributed/resource_lock.h

```diff
@@ -142,6 +144,48 @@ typedef enum CitusOperations
 						 (uint32) (taskId), \
 						 ADV_LOCKTAG_CLASS_CITUS_BACKGROUND_TASK)
 
+/*
+ * IsNodeWideObjectClass returns true if the given object class is node-wide,
+ * i.e., that is not bound to a particular database but to whole server.
+ *
+ * Defined here as an inlined function so that SET_LOCKTAG_GLOBAL_DDL_SERIALIZATION
+ * macro can use it.
+ */
+static inline bool
+IsNodeWideObjectClass(ObjectClass objectClass)
+{
+	switch (objectClass)
+	{
+		case OCLASS_DATABASE:
+		{
+			return true;
+		}
+
+		default:
+			ereport(ERROR, (errmsg("unexpected object class: %d", objectClass)));
```

Comment:
I was imagining that this function -in future- would return return false / true depending on -known- input and throw an error otherwise for safety. And given that today the only expect object class to be OCLASS_DATABASE, ended up such a code that today might not make much sense.

Do you think should we instead return false for now?

Reply:
just seen https://github.com/citusdata/citus/pull/7359#discussion_r1410803546, alright.